### PR TITLE
FortiManager Web Application Firewall Security Profile

### PR DESCRIPTION
**Pull request by ftntcorecse**
##### SUMMARY
This module allows users to manage web application firewall security profiles on a FortiManager

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
fmgr_secprof_waf.py

##### ANSIBLE VERSION

```
ansible 2.6.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/cse/PycharmProjects/JPMC_FOS/ansible/lib/ansible/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
``

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

### CODE REVIEW

**Comment by ftntcorecse** [comment]
Hi All,

May I have a status update for this PR?  Is anything else needed from me to move this forward?

Thanks

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_waf.py [hunk 1]
```diff
@@ -0,0 +1,1633 @@
...
+    description:
+      - The username associated with the account.
+    required: true
+
+  password:
+    description:
+      - The password associated with the username account.
+    required: true
+
+  mode:
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_waf.py:61]: if `mode` is similar to `state`, please update the same to `state`

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_waf.py [hunk 2]
```diff
@@ -0,0 +1,1633 @@
...
+    required: false
+    choices: ["disable", "enable"]
+
+  extended_log:
+    description:
+      - Enable/disable extended logging.
+      - choice | disable | Disable setting.
+      - choice | enable | Enable setting.
+    required: false
+    choices: ["disable", "enable"]
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_waf.py:88]:
```suggestion
    choices: 
    - disable
    - enable
```

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_waf.py [hunk 3]
```diff
@@ -0,0 +1,1633 @@
...
+    description:
+      - Enable/disable the constraint.
+      - choice | disable | Disable setting.
+      - choice | enable | Enable setting.
+    required: false
+    choices: ["disable", "enable"]
+
+  constraint_hostname_action:
+    description:
+      - Action.
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_waf.py]: Description is very generic, please update the `description` wherever needed. Also, need not to  mention choice in `description` field as you already mentioning the same in `choice` field

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_waf.py [hunk 4]
```diff
@@ -0,0 +1,1633 @@
...
+
+
+###############
+# START METHODS
+###############
+
+
+def fmgr_waf_profile_addsetdelete(fmg, paramgram):
+    """
+    fmgr_waf_profile -- Your Description here, bruh
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_waf.py]: Please update this generic comment

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_waf.py (hunk 1 above)
Comment (ftntcorecse) [reply lib/ansible/modules/network/fortimanager/fmgr_secprof_waf.py:61]: Hi @justjais mode is different than state as we use mode for the HTTP action associated with th e playbook task.  Thanks for looking at our module!

**Comment by ftntcorecse** [comment]
Hi @justjais I've added fixes to your comments above. Thanks!

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_waf.py [hunk 5]
```diff
@@ -29,9 +29,9 @@
 module: fmgr_secprof_waf
 version_added: "2.8"
 author:
-    - Luke Weighall (@lweighall)
-    - Andrew Welsh (@Ghilli3)
-    - Jim Huber (@p4r4n0y1ng)
+    - Luke Weighall (lweighall)
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_waf.py]: @ftntcorecse was there any specific reason for removing the `@` as ansible modules follow author with `@` sign.

**Review by justjais — APPROVED** [review]

**Pull request merged**